src/enclave/rpc_sessions.h

```diff
@@ -77,6 +89,24 @@ namespace enclave
       }
     };
 
+    tls::ConnID get_next_id()
+    {
+      auto id = next_client_session_id--;
+
+      if (next_client_session_id > 0)
+        next_client_session_id = -1;
+
+      while (sessions.find(id) != sessions.end())
```

Comment:
Could this loop forever in the (very unlikely) case that all outgoing ids are already taken?

Reply:
Yes, it could. It is based on the similar loop in the host-side `RPCConnections::get_next_id()`. This can only happen if we have 2^63 sessions in one of these maps, at which point we have bigger problems. But I've added a throw in both these functions, just for clarity.